crates/crypto/src/hpke.rs

```diff
@@ -584,45 +593,71 @@ where
         })
     }
 
-    const HPKE_CTX: Context = Context {
-        domain: "HPKE-v1",
-        suite_ids: &Self::HPKE_SUITE_ID,
-    };
-
     /// Performs `LabeledExtract`.
-    fn labeled_extract(salt: &[u8], label: &'static str, ikm: &[u8]) -> Prk<F::PrkSize> {
+    fn labeled_extract<'a>(
+        salt: &[u8],
+        label: &'static [u8],
+        ikm: impl IntoIterator<Item = &'a [u8]>,
+    ) -> Prk<F::PrkSize> {
         // def LabeledExtract(salt, label, ikm):
         //     labeled_ikm = concat("HPKE-v1", suite_id, label, ikm)
         //     return Extract(salt, labeled_ikm)
-        Self::HPKE_CTX.labeled_extract::<F>(salt, label, ikm)
+        let labeled_ikm = iter::once(Self::DOMAIN)
+            .chain(iter::once(Self::HPKE_SUITE_ID))
+            .chain(iter::once(label))
+            .chain(ikm);
```

Comment:
An array iter would be a little nicer. Wonder how the perf compares.

```suggestion
        let labeled_ikm = [Self::DOMAIN, Self::HPKE_SUITE_ID, label]
            .into_iter()
            .chain(ikm);
```

Reply:
Perf _should_ be the same, at least for methods that use `fold`. It's actually one word smaller.